core-java/src/main/java/org/spine3/server/RepositoryBase.java

```diff
@@ -47,16 +49,29 @@ protected RepositoryBase() {
     }
 
     private Constructor<E> getEntityConstructor() {
-        final Constructor<E> result;
+        final Class<E> entityClass = getEntityClass();
+        final Class<I> idClass = getIdClass();
         try {
-            final Class<E> entityClass = getEntityClass();
-            final Class<I> idClass = getIdClass();
+            final Constructor<E> result = entityClass.getDeclaredConstructor(idClass);
+            checkConstructorAccessModifier(result.getModifiers(), entityClass.getSimpleName());
+            return result;
+        } catch (NoSuchMethodException ignored) {
+            throw noSuchConstructorException(entityClass.getSimpleName());
+        }
+    }
 
-            result = entityClass.getConstructor(idClass);
-        } catch (NoSuchMethodException e) {
-            throw propagate(e);
+    private static void checkConstructorAccessModifier(int modifiers, String entityClassName) {
+        if (!isPublic(modifiers)) {
+            final String constructorModifier = isPrivate(modifiers) ? "private." : "protected.";
+            final String message = "Constructor must be public in the " + entityClassName + " class, not " + constructorModifier;
+            throw propagate(new IllegalAccessException(message));
         }
-        return result;
+    }
+
+    private static RuntimeException noSuchConstructorException(String entityClassName) {
+        final String message = entityClassName +
+                " class must declare a constructor with a single ID parameter (like in Aggregate class).";
```

Comment:
The part "like in Aggregate class" doesn't sound clear. What did you mean to say?